stock_picking_validate_jobcan_wf/models/stock_picking.py

```diff
@@ -56,40 +57,90 @@ def get_api_key(self, config):
             api_key = "Token " + api_key
         return api_key
 
+    def _check_jobcan_wf_status(self):
+        self.ensure_one()
+        results = []
+        if not self.jobcan_wf_number:
+            raise UserError(_("Jobcan Workflow Number missing: %s") % (self.name))
+        try:
+            params = {"id": self.jobcan_wf_number}
+            response = self.make_api_call(
+                "jobcan_wf", endpoint="v2/requests", params=params
+            )
+            response.raise_for_status()
+            results = response.json().get("results", [])
+        except Exception as e:
+            _logger.error(
+                "API call failed for picking %s with error: %s", self.id, str(e)
+            )
+            raise UserError(_("API call failed: {}".format(str(e)))) from e
+        if not results:
+            raise UserError(
+                _(
+                    "Specified Jobcan Workflow Number '%(jobcan_wf_number)s' does "
+                    "not exist: %(pick_name)s",
+                    jobcan_wf_number=self.jobcan_wf_number,
+                    pick_name=self.name,
+                )
+            )
+        if results[0].get("status") != "completed":
+            raise UserError(
+                _(
+                    "Jobcan workflow '%(jobcan_wf_number)s' is not completed: "
+                    "%(pick_name)s",
+                    jobcan_wf_number=self.jobcan_wf_number,
+                    pick_name=self.name,
+                )
+            )
+
+    def notify_user(self, message):
+        self.ensure_one()
+        if self.user_id:
+            odoobot_id = self.env["ir.model.data"]._xmlid_to_res_id("base.partner_root")
+            channel_info = self.env["mail.channel"].channel_get(
+                [odoobot_id, self.user_id.partner_id.id]
+            )
+            channel = self.env["mail.channel"].browse(channel_info["id"])
+            self_url = f"/web#id={self.id}&model=stock.picking&view_type=form"
+            message_body = _(
+                "JobCan WF ID %(wf_id)s has been approved but picking "
+                '<a href="%(url)s">%(name)s</a> failed to confirm.<br/>Details:<br/>%(message)s'
+            ) % {
+                "wf_id": self.jobcan_wf_number,
+                "url": self_url,
+                "name": self.name,
+                "message": message,
+            }
+            subject = _("Picking Confirmation Failed: %(name)s") % {"name": self.name}
+            channel.message_post(
+                body=message_body,
+                subject=subject,
+                message_type="comment",
+                subtype_xmlid="mail.mt_comment",
+                partner_ids=[self.user_id.partner_id.id],
+            )
```

Comment:
@AungKoKoLin1997 Can you please help switch to add a message in the document instead of in a channel.  Kurata-san agrees that it's better to make it visible in the document.  No need to manipulate with document followers.